### PR TITLE
Dispensing qty ds 4366

### PR DESCRIPTION
**Pull request by mprittie**
Update local `DISPENSING` CDM ETL to include `RX_QUANTITY` and `RX_DAYS_SUPPLY` values.

### CODE REVIEW

File: Oracle/PCORNetLoader_ora.sql [hunk 1]
```diff
@@ -1748,27 +1748,57 @@ PMN_EXECUATESQL(sqltext);
...
 /* Below is the Cycle 2 fix for populating the DISPENSING table  */
-select distinct
-  ibf.patient_num patid,
+with disp_status as (
+  select ibf.patient_num, ibf.encounter_num, ibf.concept_cd, ibf.instance_num, ibf.start_date, ibf.modifier_cd
+  from i2b2fact ibf
+  join BLUEHERONMETADATA.pcornet_med pnm
+    on ibf.modifier_cd=pnm.c_basecode
+  where pnm.c_fullname like '\PCORI_MOD\RX_BASIS\DI\%'
+    and length(substr(ibf.concept_cd, 5)) < 12
```
Comment (dckc) [inline Oracle/PCORNetLoader_ora.sql]: where do the numbers 5 and 12 come from?

File: Oracle/PCORNetLoader_ora.sql [hunk 2]
```diff
@@ -1748,27 +1748,57 @@ PMN_EXECUATESQL(sqltext);
...
+  ,DISPENSE_AMT  -- modifier nval_num
 --    ,RAW_NDC
 )
 /* Below is the Cycle 2 fix for populating the DISPENSING table  */
-select distinct
-  ibf.patient_num patid,
+with disp_status as (
+  select ibf.patient_num, ibf.encounter_num, ibf.concept_cd, ibf.instance_num, ibf.start_date, ibf.modifier_cd
+  from i2b2fact ibf
+  join BLUEHERONMETADATA.pcornet_med pnm
```
Comment (dckc) [inline Oracle/PCORNetLoader_ora.sql]: `BLUEHERONMETADATA` shouldn't be hard-coded; use `"&&i2b2_data_schema"`.

File: Oracle/PCORNetLoader_ora.sql [hunk 3]
```diff
@@ -1748,27 +1748,57 @@ PMN_EXECUATESQL(sqltext);
...
+    on ibf.modifier_cd=pnm.c_basecode
+  where pnm.c_fullname like '\PCORI_MOD\RX_BASIS\DI\%'
+    and length(substr(ibf.concept_cd, 5)) < 12
+)
+, disp_quantity as (
+  select ibf.patient_num, ibf.encounter_num, ibf.concept_cd, ibf.instance_num, ibf.start_date, ibf.modifier_cd, ibf.nval_num
+  from i2b2fact ibf
+  join BLUEHERONMETADATA.pcornet_med pnm
```
Comment (dckc) [inline Oracle/PCORNetLoader_ora.sql]: `"&&i2b2_data_schema"` again

File: Oracle/PCORNetLoader_ora.sql [hunk 4]
```diff
@@ -1748,27 +1748,57 @@ PMN_EXECUATESQL(sqltext);
...
+    on ibf.modifier_cd=pnm.c_basecode
+  where pnm.c_fullname like '\PCORI_MOD\PCORI_MOD\RX_QUANTITY\%'
+)
+, disp_supply as (
+  select ibf.patient_num, ibf.encounter_num, ibf.concept_cd, ibf.instance_num, ibf.start_date, ibf.modifier_cd, ibf.nval_num
+  from i2b2fact ibf
+  join BLUEHERONMETADATA.pcornet_med pnm
```
Comment (dckc) [inline Oracle/PCORNetLoader_ora.sql]: `"&&i2b2_data_schema"` again

File: Oracle/PCORNetLoader_ora.sql [hunk 5]
```diff
@@ -1759,22 +1759,22 @@ insert into dispensing (
 /* Below is the Cycle 2 fix for populating the DISPENSING table  */
 with disp_status as (
   select ibf.patient_num, ibf.encounter_num, ibf.concept_cd, ibf.instance_num, ibf.start_date, ibf.modifier_cd
-  from i2b2medfact ibf
+  from i2b2fact ibf
```
Comment (dckc) [inline Oracle/PCORNetLoader_ora.sql]: @njgraham any thoughts on why this didn't work with `i2b2medfact`?

**Review by dckc — APPROVED** [review]
I hope you can ask Nathan about the `i2b2medfact` thing and fix the hard-coded schema. But I'm not making those critical.

File: Oracle/PCORNetLoader_ora.sql (hunk 2 above)
Comment (mprittie) [reply Oracle/PCORNetLoader_ora.sql]: Good catch on forgetting to swap the `"&&i2b2_meta_schema"` schema reference back in.

**Comment by mprittie** [comment]
Thanks for the review @dckc.  I will fix the metadata schema references, and add some more inline documentation explaining hard coded values.

The reason that I didn't use the `i2b2medfact` table is because it doesn't contain NDC facts, just MEDICATION_IDs.

**Pull request merged**